### DISCUSSION TITLE
Is digital input debounced? Looking for explanation for long lick durations

### DISCUSSION THREAD

**Discussion by margo-tje** [pyPhotometry/code#21] [Q&A] [answered]
Hi,
I am trying to use a MedAssociates lickometer together with PyPhotometry (Retractable Sipper w/Lixit + Bottle for rats). This lickometer is connected to their Lickometer Control box (ENV-250B), which outputs a 28V signal. This is fed through a 28V to TTL converter (also MedAssociates; SG-231) and this TTL signal is directly plugged into PyBoard's digital 1. This should result in a 5V logic signal whenever the animal licks, and 0V (or close to ground) when the animal is not licking.

The signal I receive is 0's and 1's as expected, but I do have some 'licks' that last > 20 seconds, which cannot be physiologically true. I am currently trying to figure out what the issue is / how I can prevent this from happening.

- I have confirmed with MedAssociates that 86 Hz (framerate of my 3 channel recording) should be sufficient to capture the licking.
- I have the feeling this only happens when the animal licks very fast, and for some reason the signal going down is not being captured. I read something on the PyControl Github about debouncing; does this happen for digital inputs in PyPhotometry too?
- I thought the rats were sometimes connecting the circuit by touching the spout without actually consuming: I have 3D printed a plastic cover for the spout which improved (shortened) the lick durations significantly, but I still have a little issue with some licks being unrealistically long.
**- My main question: can this be something to be fixed in PyPhotometry or do I need to look in different directions (leaky ground, hardware issues etc)?**

I hope I explained my issue clearly and would appreciate any input on the matter.

Thanks,
Dr. Margo Slomp
Postdoc @ Berke lab (UCSF)

**Comment by ThomasAkam** [comment, marked as answer]
Hi @margo-tje,

The digital inputs do not have any debouncing or other processing, the state of each input (high or low voltage) is just recorded at the sampling rate of the photometry signals.  One thing to check is that the grounds on the Med TTL adapter and the pyPhotometry board are connected (though assuming your connecting to the BNC connector on the MED adapter this should be fine as the BNC cable has ground and signal lines).

I suggest using an oscilloscope to look at the signals coming out from the Med system (if your lab does not already have one the PicoScope 2204A is a nice and cheap option).

**Reply by margo-tje** [reply]
Hi Thomas,
Thank you for your response. I indeed use a BNC cable to connect the MED adapter to the Py board.

Will have a look with our oscilloscope, thank you.
